src/components/ChatWindow/MessageList/MessageListScrollContainer/MessageListScrollContainer.tsx

```diff
@@ -0,0 +1,144 @@
+/* istanbul ignore file */
+import React from 'react';
+import ArrowDownwardIcon from '@material-ui/icons/ArrowDownward';
+import Button from '@material-ui/core/Button';
+import clsx from 'clsx';
+import { Message } from '@twilio/conversations/lib/message';
+import throttle from 'lodash.throttle';
+import { withStyles, WithStyles, createStyles } from '@material-ui/core/styles';
+
+const styles = createStyles({
+  messageListContainer: {
+    overflowY: 'auto',
+    flex: '1',
+  },
+  outerContainer: {
+    minHeight: 0,
+    flex: 1,
+    position: 'relative',
+  },
+  innerScrollContainer: {
+    height: '100%',
+    overflowY: 'auto',
+    padding: '0 1.2em 1em',
+  },
+  button: {
+    position: 'absolute',
+    bottom: '14px',
+    right: '2em',
+    zIndex: 100,
+    padding: '0.5em 0.9em',
+    visibility: 'hidden',
+    opacity: 0,
+    boxShadow: '0px 4px 16px rgba(18, 28, 45, 0.2)',
+    transition: 'all 0.5s ease',
+  },
+  showButton: {
+    visibility: 'visible',
+    opacity: 1,
+    bottom: '24px',
+  },
+});
+interface MessageListScrollContainerProps extends WithStyles<typeof styles> {
+  messages: Message[];
+}
+interface MessageListScrollContainerState {
+  isScrolledToBottom: boolean;
+  showButton: boolean;
+  messageNotificationCount: number;
+}
+
+/*
+ * This component is a container for the MessageList component. It enables scrolling so that users
+ * can scroll through the messages as the MessageList grows taller than this container.
+ *
+ * Note that this component is tested with Cyprus only.
+ */
```

Comment:
I think the PR is good to go, but I think these comments could be a little more descriptive. I think the PR description is a good example of what could go here.

Reply:
thank you! i also spelled "Cypress" wrong 🙃